### PR TITLE
Implement seed field on sound, and add a builder

### LINKED ISSUE #754: 1.19: Sound seeds

### PR DESCRIPTION
**Pull request by zml2008**
Fixes GH-754

### CODE REVIEW

**Review by rymiel — APPROVED** [review]

File: api/src/main/java/net/kyori/adventure/sound/Sound.java [hunk 1]
```diff
@@ -274,4 +303,112 @@ interface Emitter {
...
   }
+
+  /**
+   * A builder for sound instances.
+   *
+   * <p>Type is required, all other options are optional.</p>
+   *
+   * @since 4.12.0
+   */
+  interface Builder extends AbstractBuilder<Sound> {
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/sound/Sound.java:317]: Do we not normally instead extend Buildable/Builder rather than abstract builder?

**Review by kezz — CHANGES_REQUESTED** [review]

File: api/src/main/java/net/kyori/adventure/sound/Sound.java (hunk 1 above)
Comment (zml2008) [reply api/src/main/java/net/kyori/adventure/sound/Sound.java:317]: AbstractBuilder is a new interface replacing `Builder` -- me and @kashike are discussing a new pattern for builders that replaces `Built#toBuilder()` with `Built.built(existing)`, though we're unsure about its exact form.

Comment (kezz) [reply api/src/main/java/net/kyori/adventure/sound/Sound.java:317]: Sounds dope to me then!

**Review by kezz — APPROVED** [review]

File: api/src/main/java/net/kyori/adventure/sound/Sound.java [hunk 2]
```diff
@@ -274,4 +303,112 @@ interface Emitter {
...
+    /**
+     * The volume for this sound, indicating how far away it can be heard.
+     *
+     * <p>Default value is {@code 1}.</p>
+     *
+     * @param volume the sound volume
+     * @return this builder
+     * @since 4.12.0
+     */
+    @NotNull Builder volume(final @Range(from = 0, to = Integer.MAX_VALUE) float volume);
```
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/sound/Sound.java:382]: Thought for the future: do we want to error on values outside of these?

**Pull request merged**